Node.js: Configuring Stream Response Content Info in Custom Handlers (#1671)
cap-js/docs: node.js/best-practices.md

## Custom Streaming <Beta /> { #custom-streaming-beta }

> When returning [Media Data](../guides/providing-services#serving-media-data) from a custom `READ` or operation handler in the Node.js runtime, content information can be configured as part of the handlers result object. 

Reviewer: ```suggestion
When returning [Media Data](../guides/providing-services#serving-media-data) from a custom `READ`, `action` or `function` handler, content information can be configured as part of the handlers result object. 
```

Preferably, the handler implementation should use [`req.reply`](events#req-reply), called with an instance of [stream.Readable](https://nodejs.org/api/stream.html#class-streamreadable) and options to specify content disposition headers. 

Example:

```js
srv.on('READ', 'Books', (req, next) => {
  const readable = new Readable()
  req.reply(readable, {
    mimetype: 'image/jpeg', // > optional
    filename: 'cover.jpg', // > optional
  })
})
```

Alternatively, it is also possible to return an instance of [stream.Readable](https://nodejs.org/api/stream.html#class-streamreadable) directly and configure content disposition information by assigning relevant property values (`mimetype`, `filename`) directly to that object. 

Example:

```js
srv.on('READ', 'Books', (req, next) => {
  if (coverImageIsRequested) {
    const readable = new Readable()
    return Object.assign(readable, {
      mimetype: 'image/jpeg', // > optional
      filename: 'cover.jpg', // > optional
    })
  }
  return next()
})
```

> For compatibility, the content information can be conveyed using a result object specifying the information as it would be available if extracted from the proper cds annotations.

Reviewer 2: ```suggestion
For compatibility, convey the content information using a result object. This object specifies the information as it would appear if extracted from the appropriate CDS annotations.
```

Reviewer: sure that's also a good option

Reviewer 2: ```suggestion
If needed for compatibility reasons, convey the content information using a result object specifying the information as it would appear if extracted from the appropriate CDS annotations.
```
